gateway/server.go

```diff
@@ -406,6 +407,17 @@ func (gw *Gateway) setupGlobals() {
 		go gw.flushNetworkAnalytics(gw.ctx)
 	}
 
+	var oAuthTokensPurgeInterval time.Duration
+	if purgeInterval := gw.GetConfig().Private.OAuthTokensPurgeInterval; purgeInterval == 0 {
+		oAuthTokensPurgeInterval = time.Hour
+	} else {
+		oAuthTokensPurgeInterval = time.Second * time.Duration(purgeInterval)
+	}
+
+	oauthTokensPurger := scheduler.NewScheduler("purge-oauth-tokens", oAuthTokensPurgeInterval,
+		log, gw.purgeLapsedOAuthTokens)
+	go oauthTokensPurger.Exec(gw.ctx)
```

Comment:
This may not be the appropriate place. Put it next to pubsubloop or similar goroutine creation. Also, can we use `Start` for the naming